grails-app/domain/org/pih/warehouse/invoice/InvoiceItem.groovy

```diff
@@ -97,35 +97,34 @@ class InvoiceItem implements Serializable {
         createdBy(nullable: true)
     }
 
-    ShipmentItem getShipmentItem() {
-        return shipmentItems ? shipmentItems?.iterator()?.next() : null
+    OrderItem getOrderItem() {
+        return orderItems ? orderItems?.find { it } : null
     }
 
-    String getShipmentNumber() {
-        return shipmentItem?.shipment?.shipmentNumber
+    ShipmentItem getShipmentItem() {
+        return shipmentItems ? shipmentItems?.find { it } : null
     }
 
     OrderAdjustment getOrderAdjustment() {
-        return orderAdjustments ? orderAdjustments?.iterator()?.next() : null
+        return orderAdjustments ? orderAdjustments?.find { it } : null
     }
 
-    String getOrderNumber() {
-        return shipmentItem?.orderNumber ?: orderAdjustment?.order?.orderNumber ?: null
+    Shipment getShipment() {
+        return shipmentItem?.shipment
     }
 
-    String getDescription() {
-        return orderAdjustment ? orderAdjustment.description : product?.name
-    }
-
-    OrderItem getOrderItem() {
-        if (orderAdjustments) {
-            return orderAdjustment?.orderItem
+    Order getOrder() {
```

Comment:
This method scares me the most. I think I'd prefer to leave it up to the developer unless it causes us to write this branching logic whenever we need to access Order. 

If we do need it, perhaps since a shipment might include multiple orders we always return a list of Orders here to make sure our Invoice model exposes the right method signature. It would be bad if the system supported adding multiple orders per shipment / invoice, but the model itself did not.
```
List<Order> getOrders() { 
    return orderItem ? [orderItem.order] : orderAdjustment ? [orderAdjustment.order] : shipmentItem.orderItems*.order
}
```

Reply:
I'll investigate it with other refactors, so this can be tested